src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -607,6 +608,13 @@ - (BOOL)setupCommissioningSessionWithPayload:(MTRSetupPayload *)payload
                                        error:(NSError * __autoreleasing *)error
 {
     auto block = ^BOOL {
+        // First reset previous information about this new nodeID
+        os_unfair_lock_lock(&_deviceMapLock);
+        [self _removeDeviceWithNodeID:newNodeID device:nil];
```

Comment:
The problem is that the "node ID" in this API is not an actual operational node ID.  It does not have to match the operational node ID that the commissionee will get.  It's really a "commissioning process handle" or some such...

Some API consumers might have this match actual operational node IDs.  Others will not.  The latter will be broken by these changes, as far as I can tell.